physicsnemo/models/domino/model.py

```diff
@@ -29,17 +29,34 @@
 import torch.nn.functional as F
 
 from physicsnemo.models.layers.ball_query import BallQueryLayer
+from physicsnemo.utils.profiling import profile
 
 
 def fourier_encode(coords, num_freqs):
     """Function to caluculate fourier features"""
     # Create a range of frequencies
-    freqs = torch.exp(torch.linspace(0, math.pi, num_freqs))
+    freqs = torch.exp(torch.linspace(0, math.pi, num_freqs, device=coords.device))
```

Comment:
Avoid CPU device creation.

Reply:
A little, for single device.  There's already sync points in building the hashmap for ball query, so it's not a crazy hit here.

In domain parallel mode it is brutal for performance!